File: apps/backend/app/schemas/mandi_directory.py
from __future__ import annotations

from typing import List, Optional

from pydantic import BaseModel


class MandiContact(BaseModel):
    person: Optional[str] = None
    phone: Optional[str] = None
    email: Optional[str] = None


class MandiDirectoryItem(BaseModel):
    mandi_id: str
    name: str
    state: str
    district: Optional[str] = None
    timings: Optional[str] = None
    facilities: List[str] = []
    contact: Optional[MandiContact] = None
    major_commodities: List[str] = []
    transport_info: Optional[str] = None

File: apps/frontend/src/services/mandiDirectory.ts
import api, { ApiResponse, unwrap } from "./api";

export interface MandiContact {
  person?: string;
  phone?: string;
  email?: string;
}

export interface MandiDirectoryItem {
  mandi_id: string;
  name: string;
  state: string;
  district?: string;
  timings?: string;
  facilities: string[];
  contact?: MandiContact;
  major_commodities: string[];
  transport_info?: string;
}

export interface MandiDirectoryFilters {
  state?: string;
  district?: string;
  mandi?: string;
  commodity?: string;
  limit?: number;
}

export const fetchMandiDirectory = async (filters: MandiDirectoryFilters): Promise<MandiDirectoryItem[]> => {
  const response = await api.get<ApiResponse<MandiDirectoryItem[]>>("/mandi-directory", { params: filters });
  return unwrap(response.data);
};

export const fetchMandiProfile = async (mandiId: string): Promise<MandiDirectoryItem> => {
  const response = await api.get<ApiResponse<MandiDirectoryItem>>(`/mandi-directory/${mandiId}`);
  return unwrap(response.data);
};
